src/Tasks/ResolveSDKReference.cs

```diff
@@ -1235,10 +996,10 @@ internal void AddResolutionWarning(string resourceId, params object[] parameters
             /// <summary>
             /// Get a piece of metadata off an item and make sureit is trimmed
             /// </summary>
-            private string GetItemMetadataTrimmed(ITaskItem item, string metadataName)
+            private static string GetItemMetadataTrimmed(ITaskItem item, string metadataName)
             {
                 string metadataValue = item.GetMetadata(metadataName);
-                return metadataValue = metadataValue != null ? metadataValue.Trim() : metadataValue;
+                return metadataValue != null ? metadataValue.Trim() : metadataValue;
```

Comment:
```c#
return metadataValue?.Trim();
```
?

Reply:
Isn't that redundant, since `Trim()` shouldn't ever transform a non-null input into a null output?

Reply:
Trim() is an instance method on class String, not an extension method - the "." in .Trim() would NRE

Reply:
But the null-conditional operator should take care of that, right? `metadataValue` may be `null`, in which case `metadataValue?.Trim()` would be `null`. Or it can be non-`null`, in which case `metadataValue?.Trim()` may also be non-null (it could now be `string.Empty`).

If it's `null`, the R# suggestion would map to

```c#
return null ?? null;
```

Or am I missing something? I don't think the extension-method nature of `Trim()` is relevant here.

Reply:
Just tried it in LINQpad… you're right. R# doesn't go this far though. Interesting. Updated to yours.

Reply:
Correction: R# when doing an auto-convert added the ?? but after that conversion it shows a greyed out ?? metadataValue indicating it's superfluous. So 2 steps...